packages/device_info_plus/device_info_plus/macos/device_info_plus/Sources/device_info_plus/DeviceIdentifiers.swift

```diff
@@ -0,0 +1,99 @@
+
+func getMacModelName(modelNumber: String) -> String {
+    switch modelNumber {
+    // MacBook models (2015 and later)
+    case "MacBook8,1": return "MacBook (12-inch, 2015)"
+    case "MacBook9,1": return "MacBook (12-inch, 2016)"
+    case "MacBook10,1": return "MacBook (12-inch, 2017)"
+
+    // MacBook Air models (2013 and later)
+    case "MacBookAir6,1": return "MacBook Air (11-inch, 2013)"
+    case "MacBookAir6,2": return "MacBook Air (13-inch, 2013)"
+    case "MacBookAir7,1": return "MacBook Air (11-inch, 2015)"
+    case "MacBookAir7,2": return "MacBook Air (13-inch, 2015-2017)"
+    case "MacBookAir8,1": return "MacBook Air (13-inch, 2018)"
+    case "MacBookAir8,2": return "MacBook Air (13-inch, 2019)"
+    case "MacBookAir9,1": return "MacBook Air (13-inch, 2020)"
+    case "MacBookAir10,1": return "MacBook Air (13-inch, 2020)"
+    case "Mac14,2": return "MacBook Air (13-inch, 2022)"
+    case "Mac14,15": return "MacBook Air (15-inch, 2023)"
+    case "Mac15,12": return "MacBook Air (13-inch, 2024)"
+    case "Mac15,13": return "MacBook Air (15-inch, 2024)"
+
+    // MacBook Pro models (2012 and later)
+    case "MacBookPro10,1": return "MacBook Pro (15-inch, 2012-2013)"
+    case "MacBookPro10,2": return "MacBook Pro (13-inch, 2012-2013)"
+    case "MacBookPro11,1": return "MacBook Pro (13-inch, 2013-2014)"
+    case "MacBookPro11,2", "MacBookPro11,3": return "MacBook Pro (15-inch, 2013-2014)"
+    case "MacBookPro11,4", "MacBookPro11,5": return "MacBook Pro (15-inch, 2015)"
+    case "MacBookPro12,1": return "MacBook Pro (13-inch, 2015)"
+    case "MacBookPro13,1": return "MacBook Pro (13-inch, 2016)"
+    case "MacBookPro13,2": return "MacBook Pro (13-inch, 2016)"
+    case "MacBookPro13,3": return "MacBook Pro (15-inch, 2016)"
+    case "MacBookPro14,1": return "MacBook Pro (13-inch, 2017)"
+    case "MacBookPro14,2": return "MacBook Pro (13-inch, 2017)"
+    case "MacBookPro14,3": return "MacBook Pro (15-inch, 2017)"
+    case "MacBookPro15,1", "MacBookPro15,3": return "MacBook Pro (15-inch, 2018-2019)"
+    case "MacBookPro15,2": return "MacBook Pro (13-inch, 2018-2019)"
+    case "MacBookPro15,4": return "MacBook Pro (13-inch, 2019)"
+    case "MacBookPro16,1", "MacBookPro16,4": return "MacBook Pro (16-inch, 2019)"
+    case "MacBookPro16,2": return "MacBook Pro (13-inch, 2019)"
+    case "MacBookPro16,3": return "MacBook Pro (13-inch, 2020)"
+    case "MacBookPro17,1": return "MacBook Pro (13-inch, 2020)"
+    case "MacBookPro18,1": return "MacBook Pro (14-inch, 2021)"
+    case "MacBookPro18,2": return "MacBook Pro (16-inch, 2021)"
+    case "MacBookPro18,3": return "MacBook Pro (16-inch, 2021)"
+    case "Mac14,5", "Mac14,9": return "MacBook Pro (14-inch, 2023)"
+    case "Mac14,6", "Mac14,10": return "MacBook Pro (14-inch, 2023)"
+    case "Mac14,7": return "MacBook Pro (13-inch, 2022)"
+    case "Mac15,3": return "MacBook Pro (14-inch, 2023)"
```

Comment:
For these lists I wasn't sure if we need to specify CPU model, like it is done on [Apple's website](https://support.apple.com/en-us/102869) but decided that it is redundant info.